### DISCUSSION TITLE
Get Cross-Reference Clipping Boundaries

### DISCUSSION THREAD

**Discussion by treynohren1995** [mozman/ezdxf#828] [Q&A] [answered]
Hello! I have a file (XREF.dxf) that is cross-referenced in another drawing (INPUT_DRAWING.dxf). I'd like to copy the cross-reference from INPUT_DRAWING.dxf to an OUTPUT_DRAWING.dxf. As of now, I'm able to copy the block definition for the cross-referenced file, and add references to that block definition on the new file. However, I haven't been able to figure out how to get the clipping boundaries of those references. For instance:
XREF.dxf's original boundaries are something as follows:
```
------------------
|                |
|                |
------------------
```
The cross-reference in INPUT_DRAWING.dxf has been clipped to something smaller:
```
----------
|        |
----------
```
In the generated OUTPUT_DRAWING.dxf, the original XREF.dxf boundaries have been kept.
```
------------------
|                |
|                |
------------------
```

Any ideas on where within INSERTS or block definitions I can find those clipping boundaries? Thank you!

Also for reference, I've tried opening INPUT_DRAWING.dxf and immediately saving as OUTPUT_DRAWING.dxf through ezdxf. This does keep the boundaries of the xref defined in INPUT_DRAWING.dxf, so I know that the clipped boundaries are probably stored somewhere.

Thank you!

**Comment by mozman** [comment, marked as answer]
I assume you are using the `Importer` add-on to import data from the XREF.dxf file, which is not capable to import the clipping path of block references (INSERT entities). The clipping path is a complex feature and not documented by the DXF reference, but support for this feature is on my TODO list see #760.

I am currently working on a new module called `xref` which is much more powerful than the `Importer` add-on for loading (not importing) data from external DXF files (XREFs). The import process of the `Importer` add-on removes most external dependencies such as XDATA, custom app-data and all links to objects in the OBJECTS section, the loading process of the `xref` module preserves most if not all of this data.

**Reply by treynohren1995** [reply]
Ahh, that makes a lot of sense. Thank you very much for the explanation!
